aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncMocking.java

```diff
@@ -213,91 +224,159 @@ public static OnDeleteConfigurator onDelete(AppSync mock) {
+                    new GraphQLResponse<>(
+                            new PaginatedResult<>(items, requestForNextResult),
+                            Collections.emptyList()
+                    )
+            );
+        }
+
+        /**
+         * Configures an instance of an {@link AppSync} to invoke the response callback with the provided mockResponse
+         * when asked to {@link AppSync#sync(GraphQLRequest, Consumer, Consumer)}.
+         * @param requestMatcher ArgumentMatcher which returns true if the GraphQLRequest should be mocked.
+         * @param mockResponse GraphQLResponse to be passed back in the response callback.
+         * @param <M> Type of models for which a response is mocked
+         * @return The same Configurator instance, to enable chaining of calls
+         */
+        public <M extends Model> SyncConfigurator mockSuccessResponse(
+                ArgumentMatcher<GraphQLRequest<PaginatedResult<ModelWithMetadata<M>>>> requestMatcher,
+                GraphQLResponse<PaginatedResult<ModelWithMetadata<M>>> mockResponse) {
             doAnswer(invocation -> {
                 // Get a handle to the response consumer that is passed into the sync() method
-                // Response consumer is the third param, at index 2 (@0, @1, @2, @3).
-                final int argumentPositionForResponseConsumer = 2;
-                final Consumer<GraphQLResponse<Iterable<ModelWithMetadata<T>>>> consumer =
+                // Response consumer is the second param, at index 1 (@0, @1, @2).
+                final int argumentPositionForResponseConsumer = 1;
+                final Consumer<GraphQLResponse<PaginatedResult<ModelWithMetadata<M>>>> consumer =
                     invocation.getArgument(argumentPositionForResponseConsumer);
 
-                // Call the response consumer, and pass the mocked items
-                // inside of a GraphQLResponse wrapper
-                final Iterable<ModelWithMetadata<T>> data = new HashSet<>(Arrays.asList(responseItems));
-                consumer.accept(new GraphQLResponse<>(data, Collections.emptyList()));
+                // Call the response consumer, and pass the mocked response
+                consumer.accept(mockResponse);
 
                 // Return a NoOp cancelable via the sync() method's return.
                 return new NoOpCancelable();
             }).when(appSync).sync(
-                eq(modelClass), // Item class to sync
-                any(), // last sync time
-                any(), // Consumer<Iterable<ModelWithMetadata<T>>>
-                any() // Consumer<DataStoreException>
+                argThat(requestMatcher),
+                any(), // Consumer<GraphQLResponse<PaginatedResult<ModelWithMetadata<M>>>>
+                any()  // Consumer<DataStoreException>
             );
             return SyncConfigurator.this;
         }
 
+        private <M extends Model> ArgumentMatcher<GraphQLRequest<PaginatedResult<ModelWithMetadata<M>>>> matcherFor(
```

Comment:
Method name: `matchesRequest`? It'll look cool above when you write `argThat(matchesRequest(`!